src/components/molecules/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,139 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useState, FC, FormEvent, ChangeEvent } from 'react';
+
+import Button, { EButtonTheme } from '@components/atoms/Button';
+import Input from '@components/atoms/Input';
+import { ETypographyVariant } from '@enums/typography';
+import { useAuth } from '@src/contexts/AuthContext';
+import { Auth } from '@types';
+
+import * as S from './LoginForm.styled';
+
+interface LoginProps {
+  type: string;
+}
```

Comment:
I'd make this prop an enum:
```suggestion
enum EAuthType {
  Login = 'LOGIN',
  Signup = 'SIGNUP'
}

interface LoginProps {
  type: EAuthType;
}
```